packages/app/components/stack-modal/StackOrdersTable.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+import { cowExplorerUrl } from "@/models/cow-order";
+import { getOrderPairSymbols } from "@/models/order";
+import { StackOrderProps } from "@/models/stack-order";
+import {
+  BodyText,
+  Icon,
+  Table,
+  TableBody,
+  TableCaption,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow
+} from "@/ui";
+import { formatDate } from "@/utils/datetime";
+import { convertedAmount } from "@/utils/numbers";
+import { addressShortner } from "@/utils/token";
+import { Order as CowOrder } from "@cowprotocol/cow-sdk";
+import Link from "next/link";
+
+const INITIAL_NUMBER_OF_COW_ORDERS = 8;
+
+export const StackOrdersTable = ({ stackOrder }: StackOrderProps) => {
+  const initialCowOrders =
+    stackOrder?.cowOrders?.slice(0, INITIAL_NUMBER_OF_COW_ORDERS) ?? [];
+
+  const [cowOrders, setCowOrders] = useState<CowOrder[]>(initialCowOrders);
+
+  function addMoreOrders() {
+    setCowOrders(stackOrder.cowOrders.slice(0, cowOrders.length + 4));
+  }
```

Comment:
Can it be an arrow function? This feels weird x)

Reply:
Also, can you name this 4 as you named the 8?

Reply:
I'm a bit of liking the `function` now, but yeah, I will turn it into a arrow fn.

Reply:
I can name the 4, do you have suggestion on the naming?

`MORE_ORDERS_NUMBER`, `NUMBER_OF_NEW_ORDERS` ??

Reply:
`MORE_ORDERS_NUMBER` seems fine for me